runtime/onert/backend/train/ops/FullyConnectedLayer.cc

```diff
@@ -134,21 +134,16 @@ void FullyConnectedLayer::backwardFloat32()
 {
   // Calculate gradient for activation
   const IPortableTensor *backprop_act;
-  switch (_activation)
+  try
```

Comment:
I'm not sure if adding a try block here just for a little better message. Try blocks affect performance.

Reply:
I agree with @ragmani .

Reply:
However, the message like `train FuulyConnectedLayer` is very useful for debug/devel, I think.

Reply:
Hmm... how to achieve two goals?

Reply:
Yes.. This try-catch is only for adding `train ...Layer` in error message. 

`Try..Catch` also heart readability a bit. :thinking:

Reply:
How about wrapping try{} .. in more higher level? 

Maybe here? 

### AS-IS 

https://github.com/Samsung/ONE/blob/e50a233e60eeaa9cf37ea1cc16eb25646b088911/runtime/onert/core/src/exec/train/TrainableFnSequence.cc#L34-L40 

### TO-BE 

```cc
//TrainableSequence.cc 
void TrainableFnSequence::backward(uint32_t training_step)
{
  for (auto it = _functions.rbegin(); it != _functions.rend(); ++it)
  {
    try{
      (*it)->backward();
    }
    catch{ 
       // print layer name and detailed message here 
    }   
}

Reply:
I don't know whether higher level is more efficient. Rather, I think the currrent code is better in error handling.

Reply:
> I don't know whether higher level is more efficient.  

I thought the above approach because it could give some consistency in the error message format.
Like... Inside the `backward()`, Any function could throw an error in any way, and `catch{}` gives a kind of tag -  where it problem resides. 

Yes, As you pointed out, this way isn't efficient either. 

> Do you think it's necessary to add a try block just for branching error messages? 

I also like to remove try{} block. 
But I think adding a message to show which layer is problematic is necessary.
That message could help the user(developer) a lot - If a user finds a problematic layer(node) easily, They could take a walk-around - like removing a problematic layer from the model and retrying.

Reply:
> Is there any way to branch error message?

:thinking: I could make the function(`backpropActivation`) return nullptr when error occurs, instead of thorwing an error.
If so, caller should check the result value isn't nullptr; if result value is nullptr, caller might throw an error like before. 

Maybe caller part looks like this : 
```cpp
const IPortableTensor *backprop_act =  backpropActivation(_activation, _output, _back_prop_output, _act_back_prop_output.get());

if(backprop_act ==  nullptr)
   throw runtime_error{"train Convolution layer : error ... "};
```

Reply:
> I also like to remove try{} block.
But I think adding a message to show which layer is problematic is necessary.
That message could help the user(developer) a lot - If a user finds a problematic layer(node) easily, They could take a walk-around - like removing a problematic layer from the model and retrying.

I think developers can see which layer is problematic thought debugger. So, I'm not sure if it's worth adding a try bock.

@ragmani Could you share your opinion about this direction? :)

It looks better in performance. But, it seems unclear what message to write. I don't know there is another way to branch error messages. 🤔, Let's keep the current code and modify it later if there is a performance issue.